Add support to net_map plugin for configuring Multus routes

lewisdenny opened the pull request:
As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running

ninja-quokka committed:
Add support to net_map plugin for configuring Multus routes

lewisdenny commented on plugins/module_utils/net_map/networking_definition.py:
```diff
@@ -583,10 +583,10 @@ def get_version_from_raw(
             return None
 
-        start_version = cls.__get_field_ip_net_version(
+        start_version = cls.get_field_ip_net_version(
```
I'm not sure if this is a good idea but I didn't want to duplicate the code so I made the function public. Maybe I should be doing something else here though

pablintino replied:
It's more than fine man, but, if was you, I'd move the entire method to a plain function, cause now it has zero "direct" relation to this class but with both classes that calls it.

lewisdenny commented on plugins/module_utils/net_map/networking_definition.py:
```diff
@@ -1377,6 +1557,68 @@ def __pick_nets_from_raw(
 
         return self.__ipv4_network, self.__ipv6_network
 
+    def __pick_routes_from_raw(
```
I think I'm going to do something different here so I didn't finish this function yet.

lewisdenny commented on plugins/module_utils/net_map/networking_mapper.py:
```diff
@@ -610,7 +627,7 @@ def __build_network_tools(
         )
         multus = (
-            self.__build_network_tool_common(
+            self.__build_network_tool_route(
```
the netconfig tool also supports configuring routes so I would like to test switching it to `__build_network_tool_route` as well

lewisdenny commented on tests/unit/module_utils/test_files/networking-definition-valid-all-tools-dual-stack.yml:
```diff
@@ -17,6 +17,13 @@ networks:
+            gateway: "192.168.122.1"
+          - destination: "172.168.121.0/24"
+            gateway: "172.168.122.1"
+          # - destination: "2001:db8:3333:4444:5555:6666:7777:8888"
```
IPV6 routes basically work, I just need to use a good example

lewisdenny commented on tests/unit/module_utils/test_files/networking-definition-valid-all-tools-full-map-out.json:
```diff
@@ -365,4 +373,4 @@
   },
   "routers": {}
-}
+}
```
I will fix the absent newline at the end of these files before opening a real PR

pablintino commented on plugins/module_utils/net_map/networking_mapper.py:
```diff
@@ -650,6 +667,32 @@ def __build_network_tool_common(
+        tool_type: typing.Type,
+    ) -> networking_env_definitions.MappedMultusNetworkConfig:
+        args_list = [
+            [
+                _map_host_network_range_to_output(ip_range)
+                for ip_range in tool_net_def.ranges_ipv4
+            ],
+            [
+                _map_host_network_range_to_output(ip_range)
+                for ip_range in tool_net_def.ranges_ipv6
+            ],
```
You can create these two by calling `__build_network_tool_common`. A bit less to maintain.

pablintino commented on plugins/module_utils/net_map/networking_definition.py:
```diff
@@ -894,6 +894,129 @@ def __str__(self):
         return f"{self.__start_ip}-{self.__end_ip}"
 
 
+class HostNetworkRoute(ansible_encoding.RawConvertibleObject):
```
The `HostNetworkRange` inherits `RawConvertibleObject` because that way I can use the class as part of the exceptions we raise. If we are not going to pass instances of this class to an exception we are safe without this.
PS: I don't see where the abstract method is implemented.

cjeanner committed:
Create a new .utility DNS zone

This zone will allow to get fixed names to access utility services, such
as the IPMI.
Using names instead of IPs ensures we can easily inject IPv6 in the
game, without having to chase down IPs in the code: just update the DNS
record to get either IPv4, or IPv6, or even dual-stack.

This patch also updates sushy_emulator role to remove some hacks
injected as "quick workaround". The cifmw_sushy_emulator_listen_ip
parameter is now passed as-is and, if needed, any IPv6 we really pass
there would be properly wrapped.
In the reproducer context, it will get a hostname instead, ensuring we
pass via the .utility zone and appropriate records.

pablintino committed:
[ci_nmstate] Allow users to directly use the unprovisioned tasks

At the same time, fix a format issue in the Net Env Def
that assumes the parent_interface is always present. That's
no longer true since we added the new mapper that works in best
effort mode.

hjensas committed:
devscripts - allow overriding branch on git checkout

Use a variable for the branch when doing git checkout of devscripts.

For exameple:
  cifmw_devscripts_repo: 'https://github.com/hjensas/dev-scripts.git'
  cifmw_devscripts_repo_branch: support-ipv6-connected
